File: src/main/java/com/wesker926/leetcode/algorithms/p1518/s1/Solution.java
package com.wesker926.leetcode.algorithms.p1518.s1;

/**
 * @author wesker.gh
 * @date 2021/12/17
 * @description 模拟
 */
public class Solution {
    public int numWaterBottles(int numBottles, int numExchange) {
        int ans = numBottles, d = 0, m = 0;
        while (numBottles >= numExchange) {
            d = numBottles / numExchange;
            m = numBottles - d * numExchange;
            ans += d;
            numBottles = d + m;
        }
        return ans;
    }
}

File: src/main/java/com/wesker926/leetcode/algorithms/p1518/s1/solution.go
// Package s1
// @author: wesker.gh
// @date: 2021/12/17
// @description: 模拟
package s1

func numWaterBottles(numBottles int, numExchange int) int {
	ans, d, m := 0, 0, 0
	for numBottles >= numExchange {
		d = numBottles / numExchange
		m = numBottles - numExchange*d
		ans += numBottles - m
		numBottles = d + m
	}
	return ans + numBottles
}
